include/group/gemm/impl/default_fpu_xe.hpp

```diff
@@ -353,6 +356,21 @@ class gemm_t<
     }
 
 private:
+    inline void reorder_matA(matA_t &matA) {
+        constexpr uint32_t num_block_x = tile_size_x_a / block_size_x_a;
+        constexpr uint32_t num_block_y = tile_size_y_a / block_size_y_a;
+        for (int i = 0; i < num_block_y * num_block_x; i++) {
+            auto dst_blk = matA.reg.xetla_select<matA_t::block_elems, 1>(
+                    i * matA_t::block_elems);
+            xetla_vector<float, matA_t::block_elems> trans_blk;
+            for (int j = 0; j < block_size_y_a; j++) {
+                trans_blk.xetla_select<block_size_y_a, block_size_x_a>(j)
+                        = dst_blk.xetla_select<block_size_y_a, 1>(
+                                j * block_size_x_a);
+            }
+            dst_blk = trans_blk;
+        }
```

Comment:
Should be able to handle cases where `block_size_y_a != block_size_x_a` with the following changes

```suggestion
        for (uint32_t i = 0; i < num_block_y * num_block_x; i++) {
            auto dst_blk = matA.reg.xetla_select<matA_t::block_elems, 1>(
                    i * matA_t::block_elems);
            xetla_vector<dtype_a, matA_t::block_elems> trans_blk;
            for (uint32_t j = 0; j < block_size_y_a; j++) {
                trans_blk.xetla_select<block_size_x_a, block_size_y_a>(j)
                        = dst_blk.xetla_select<block_size_x_a, 1>(
                                j * block_size_x_a);
            }
            dst_blk = trans_blk;
        }
```